dartagnan/src/main/java/com/dat3m/dartagnan/witness/WitnessBuilder.java

```diff
@@ -155,7 +156,8 @@ public WitnessGraph build() {
 				graph.addEdge(edge);
 
 				nextNode++;
-				if (e.hasTag(Tag.ASSERTION)) {
+				if (e instanceof Assert) {
+					// FIXME: Execution of an assertion is not a problem, if the assertion holds!
```

Comment:
can you simply add `prover.getModel().evaluate(e.getExpression())`?

Reply:
Yes. But this code is true to the original code (which was already wrong) :)

Reply:
the original code was only used with boogie which always moved the expression to the control flow and generated an `assert(false)`, thus the original code was (practically) sound while this one is not.

Reply:
I don't think this is specific to boogie. LLVM generates the code such that it only has `assert(false)`. So it is still practically sound.